pkg/typesystem/typesystem.go

```diff
@@ -177,6 +182,8 @@ func New(model *openfgav1.AuthorizationModel) *TypeSystem {
 			}
 
 			tdRelations[relation] = r
+			ttus := make([]*openfgav1.TupleToUserset, 0)
```

Comment:
Are we able to know the size and preallocate instead of allocating per slice?

Reply:
Hmmm we'd have to recursively walk through the relation definition, which we do in the next line. I could preallocate a fixed size like 100, but i don't see much benefit. The problem with not preallocating is that if you add many items, Go will have to copy all elements into bigger and bigger arrays, but here, we won't have many elements. And if we do, we only pay the penalty once when building the model

Reply:
If we don't have many items, that's fine.  I was worried that if we have lots of items, there is a lot of copying / moving elements into larger array.

Reply:
I mean, we don't have a limit of number of statements in a relation definition, but I can't imagine someone writing a very long line? `define viewer as a1 or a2 or a3 .... or a100`?